### ISSUE TITLE
No matching export in "node_modules/@angular/common/fesm2022/http.mjs" for import "HttpStatusCode"

### ISSUE THREAD

**Issue by Hagriel** [angular/angular-cli#25327]
### Which @angular/* package(s) are the source of the bug?

common

### Is this a regression?

Yes

### Description

 No matching export in "node_modules/@angular/common/fesm2022/http.mjs" for import "HttpStatusCode"

    src/app/core/services/interceptors/api-token.interceptor.ts:1:23:
      1 │ import { HttpResponse, HttpStatusCode, } from '@angular/common/http';

### Please provide the environment you discovered this bug in (run `ng version`)

```true
Angular: 16.0.4
"builder": "@angular-devkit/build-angular:browser-esbuild",
```

**Comment by NatanCoralogix** [comment]
Im facing similar issues after upgrading from 16.0.3 -> 16.0.5 

for example:
No matching export in "node_modules/@angular/core/fesm2022/core.mjs" for import "SimpleChanges"

**Comment by JeanMeche** [comment]
Does this issue still happens when deleting `.angular` directory (its a cache directory) ? 

We're still waiting for a repro to be able to investigate it !

**Comment by NatanCoralogix** [comment]
@JeanMeche i dont have a .angular folder.
maybe esbuild doesn't care about it?

**Comment by Hagriel** [comment]
@JeanMeche 
for my case issue was fixed after update to 6.0.5

**Comment by JoostK** [comment]
For esbuild this is expected, as it is effectively constrained by `isolatedModules` restrictions and hence unaware of declarations.  However, TS -> JS compilation is always done using the TypeScript compiler which—unless configured with the options mentioned in [#25327 (comment)](https://github.com/angular/angular-cli/issues/25327#issuecomment-1580117209)—should elide type-only imports before esbuild even sees them.

**Comment by kemsky** [comment]
@JoostK, sorry, but what exactly should be set in tsconfig to make it work? Enabling `isolatedModules` immediately results in `TS2748: Cannot access ambient const enums when 'isolatedModules' is enabled.`.

**Comment by NatanCoralogix** [comment]
update:
i upgraded to version 16.1.0 and it kind of fix the issues - but started another weird issue:
lets say i have:

`import { IVariable, MultiSelectSelectionType, MultiSelectVariableSourceType } from '@app/features/dashboard/model/variable.model';`
i got error that`IVariable` is not exported or something, while it is:
`export interface IVariable { ...}`
the solution was to import it alone:
`import { IVariable } from '@app/features/dashboard/model/variable.model';`

after that, it compiled successfully, but got error in the browser:
`Uncaught SyntaxError: The requested module '/.angular/cache/16.1.0/vite/deps/@angular_core.js?v=091cc3f7' does not provide an export named 'SimpleChanges' (at alerts-monitoring-filters.component.ts:11:3)`
it try to find  SimpleChanges from the cache of core, but its not there.

im lost what to do here..
any idea?

**Comment by e-oz** [comment]
@NatanCoralogix I had the same error, `"optimization": true,` option helped (in project.json or angular.json).

**Comment by NachmanBerkowitz** [comment]
I experienced the same as @NatanCoralogix. Would love to know the reason for this, and if it will be fixed to work without setting "optimization": true

**Comment by Vladyslav-Dovzhenko** [comment]
Also helps replacing
`import { SimpleChanges } from '@angular/core' ` to `import type { SimpleChanges } from '@angular/core'`

**Comment by NatanCoralogix** [comment]
for me it solved with version 16.1.2
